lib/interactive_viewer.ml

```diff
@@ -103,34 +103,79 @@ let change_summary z_patches : ui Lwd.t =
   in
   W.string ~attr:Notty.A.(fg lightcyan) operation_count
 
-let view (patches : Patch.t list) =
+let content_image = Lwd.var (Notty.I.vcat [ Notty.I.string Notty.A.empty "" ])
+
+let content_image_renderer (ui : ui) =
+  let height = Ui.layout_height ui in
+  let width = Ui.layout_width ui in
+  let renderer = Renderer.make () in
+  Renderer.update renderer (width, height * 3) ui;
+  renderer
+
+let view (patches : Patch.t list) (events : Notty.Unescape.event list) =
   let z_patches : 'a Zipper.t Lwd.var =
     match Zipper.zipper_of_list patches with
     | Some z -> Lwd.var z
     | None -> failwith "zipper_of_list: empty list"
   in
-  W.vbox
+  let handle_event (event : Notty.Unescape.event) : Ui.may_handle =
+    match event with
+    | `Key (`ASCII 'q', []) ->
+        Lwd.set quit true;
+        `Handled
+    | `Key (`ASCII 'n', []) ->
+        navigate z_patches Next;
+        `Handled
+    | `Key (`ASCII 'p', []) ->
+        navigate z_patches Prev;
+        `Handled
+    | _ -> `Unhandled
+  in
+  let handle_event_ui (event : Ui.key) : Ui.may_handle =
+    match event with
+    | `ASCII 'q', [] ->
+        Lwd.set quit true;
+        `Handled
+    | `ASCII 'n', [] ->
+        navigate z_patches Next;
+        `Handled
+    | `ASCII 'p', [] ->
+        navigate z_patches Prev;
+        `Handled
+    | _ -> `Unhandled
+  in
```

Comment:
Having a duplicated handling of events is not ideal! It would be better to have a list of `Ui.key` as the `events` argument.